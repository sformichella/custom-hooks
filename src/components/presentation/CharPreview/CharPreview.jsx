import React from 'react';
import PropTypes from 'prop-types';

export default function CharPreview({ id, name, image }) {
  return (
    <>
      <a href={`/${id}`}>{name}</a>
      <img src={image}/>
    </>
  );
}

CharPreview.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};
