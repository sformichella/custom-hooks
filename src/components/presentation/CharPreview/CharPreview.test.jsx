import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharPreview from './CharPreview';

describe('CharPreview component', () => {
  afterEach(() => cleanup());
  it('renders CharPreview', () => {
    const { asFragment } = render(<CharPreview />);
    expect(asFragment()).toMatchSnapshot();
  });
});
