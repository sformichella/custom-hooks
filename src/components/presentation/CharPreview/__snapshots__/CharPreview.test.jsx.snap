// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharPreview component renders CharPreview 1`] = `
<DocumentFragment>
  <a
    href="/1"
  >
    Rick Sanchez
  </a>
  <img
    src="rick-sanchez.com"
  />
</DocumentFragment>
`;
